workspaces/adventure-pack/goodies/typescript/Iterator.prototype.toSet/index.test.ts

```diff
@@ -0,0 +1,113 @@
+import { describe, expect, it } from "@jest/globals";
+
+import "../Iterator.prototype.filter";
+import "../Iterator.prototype.map";
+
+import { iteratorPrototype } from "../Iterator.prototype";
+delete (iteratorPrototype as unknown as Record<string, unknown>).toSet;
+// eslint-disable-next-line import-x/first -- This has to happen after we delete the built-in implementation.
+import "./index";
+
+describe("Iterator.prototype.toSet", () => {
+  it("adds primitives to a Set", () => {
+    const arr = [1, 2, 2, 3, 1, 2, 3, 3, 100, false, 100, false, "hello"];
+    const set = arr.values().toSet();
+
+    expect(set).toStrictEqual(new Set([1, 2, 3, 100, false, "hello"]));
+  });
+
+  it("adds objects to a Set", () => {
+    const obj1 = { a: 1 };
+    const obj2 = { b: 2 };
+    const objectArray = [obj1, obj2, obj1, obj2, obj2, obj1];
+    const set = objectArray.values().toSet();
+
+    expect(set).toStrictEqual(new Set([obj1, obj2]));
+  });
+
+  it("adds arrays to a Set", () => {
+    const arr1 = [1, 2];
+    const arr2 = [3, 4];
+    const arrayOfArrays = [
+      arr1,
+      arr2,
+      arr1,
+      arr1,
+      arr1,
+      arr1,
+      arr1,
+      arr1,
+      arr2,
+      arr2,
+      arr2,
+    ];
+    const set = arrayOfArrays.values().toSet();
+
+    expect(set).toStrictEqual(new Set([arr1, arr2]));
+  });
+
+  it("can convert multiple Generators to a Set", () => {
+    const generator1 = function* (): Generator<string, void, void> {
```

Comment:
I don't think we've been super consistent about naming across tests, but this object is technically not a generator, it's a _generator factory_, i.e. a function that creates a new generator whenever it's invoked.

I'm slightly tempted to turn these into IIFEs so they're actually generators, but I don't think it's a big deal either way, so up to you.

Reply:
ok. I think I will leave them as they are for now.